lib/instrumentation/modules/http2.js

```diff
@@ -68,6 +77,60 @@ module.exports = function (http2, agent, { enabled }) {
         eos(stream, function () {
           trans.end()
         })
+      } else if (event === 'request' && reqIsHTTP1(stream)) {
+        // http2.createSecureServer() supports a `allowHTTP1: true` option.
+        // When true, an incoming client request that supports HTTP/1.x but not
+        // HTTP/2 will be allowed. It will result in a 'request' event being
+        // emitted. We wrap that here.
+        //
+        // Note that, a HTTP/2 request results in a 'stream' event (wrapped
+        // above) *and* a 'request' event. We do not want to wrap the
+        // compatibility 'request' event in this case. Hence the `reqIsHTTP1`
+        // guard.
+        const req = stream
+        const res = headers
+
+        agent.logger.debug('intercepted request event call to http2.Server.prototype.emit for %s', req.url)
+
+        var traceparent = req.headers.traceparent || req.headers['elastic-apm-traceparent']
```

Comment:
No changes needed here -- but it's worth pointing out that the existing HTTP2 stream instrumentation does not appear to support distributed tracing at all.  Open issue for that here: https://github.com/elastic/apm-agent-nodejs/issues/1830

Reply:
Thanks. Yes, I noticed that issue when adding other http2-related deficiency issues.